src/Scripts/Game/Features/Gathering/EL_GatherAction.c

```diff
@@ -5,31 +5,114 @@ class EL_GatherAction : ScriptedUserAction
 	
 	[Attribute("", UIWidgets.ResourceNamePicker, desc: "Prefab what item is gathered")]
 	private ResourceName m_GatherItemPrefab;
+	
+	[Attribute(defvalue:"1", UIWidgets.EditBox, desc: "Amount of items to receive")]
+	private int m_AmountGathered;
+	
+	[Attribute("", UIWidgets.EditBox, desc: "Display name of what is needed")]
+	private string m_RequiredItemDisplayName;
+	
+	[Attribute("", UIWidgets.ResourceNamePicker, desc: "Item(s) required for gathering")]
+	private ResourceName m_RequiredItemPrefab;
+	
+	[Attribute("", UIWidgets.CheckBox, desc: "Check entire inventory too")]
+	private bool m_CheckInventory;
 		
+	[Attribute(defvalue:"0", UIWidgets.EditBox, desc: "Amount of delay after performing action")]
+	private int m_DelayTimeMilliseconds;
```

Comment:
What is the idea behind the delay? If it is to limit the gathering speed, then that is instead controlled by how long the action takes isn't it?

Reply:
One use I thought of for a delay would be something like robbing a trader. You wouldn't want a user to be able to spam the gather function which would require a separate timer from the duration one.

Another possibility I was thinking of adding was the ability to toggle the entities visibility while it's on delay so that it is hidden, but reappears.